pkg/pullexecutor/pullexecutor.go

```diff
@@ -102,7 +102,8 @@ func (m *PullExecutor) StartPulling(o *PullOptions) error {
 			c, cancel := context.WithTimeout(context.Background(), pullCtxTimeout)
 			defer cancel()
 
-			if pullstatus.Get(o.NamedRef) == pullstatus.StillPulling {
+			if pullstatus.Get(o.NamedRef) == pullstatus.StillPulling ||
+				pullstatus.Get(o.NamedRef) == pullstatus.Pulled {
```

Comment:
This is a fix for what appears to be a race condition when I was testing https://github.com/warm-metal/csi-driver-image/pull/81
Basically when two requests sent back-to-back with little time duration between them, one of them succeeds while the other goes into the mutex section and continues to pull the same image again (if `pullAlways: true`). I haven't seen this issue outside https://github.com/warm-metal/csi-driver-image/pull/81 but thought it would be good to get the fix in before we get https://github.com/warm-metal/csi-driver-image/pull/81 merged.

Reply:
Good catch!